src/aspire/sinogram/sinogram.py

```diff
@@ -0,0 +1,140 @@
+                f"Invalid data shape: {data.shape}. Expected shape: (..., angles, radial_points), where '...' is the stack number."
+            )
+        self._data = data.astype(self.dtype, copy=False)
+        self.ndim = self._data.ndim
+        self.shape = self._data.shape
+        self.stack_shape = self._data.shape[:-2]
+        self.stack_n_dim = self._data.ndim - 2
+        self.n = np.product(self.stack_shape)
+        self.n_angles = self._data.shape[-2]
+        self.n_radial_points = self._data.shape[-1]
+
+        # Numpy interop
+        # https://numpy.org/devdocs/user/basics.interoperability.html#the-array-interface-protocol
+        self.__array_interface__ = self._data.__array_interface__
+        self.__array__ = self._data
+
+    def _check_key_dims(self, key):
+        if isinstance(key, tuple) and (len(key) > self._data.ndim):
+            raise ValueError(
+                f"Sinogram stack_dim is {self.stack_n_dim}, slice length must be =< {self.n_dim}"
+            )
+
+    def __getitem__(self, key):
+        self._check_key_dims(key)
+        return self.__class__(self._data[key])
+
+    def __setitem__(self, key, value):
+        self._check_key_dims(key)
+        self._data[key] = value
+
+    def stack_reshape(self, *args):
+        """
+        Reshape the stack axis.
+
+        :*args: Integer(s) or tuple describing the intended shape.
+
+        :returns: Sinogram instance
+        """
+
+        # If we're passed a tuple, use that
+        if len(args) == 1 and isinstance(args[0], tuple):
+            shape = args[0]
+        else:
+            # Otherwise use the variadic args
+            shape = args
+
+        # Sanity check the size
+        if shape != (-1,) and np.prod(shape) != self.n:
+            raise ValueError(
+                f"Number of sinogram images {self.n} cannot be reshaped to {shape}."
+            )
+
+        return self.__class__(self._data.reshape(*shape, *self._data.shape[-2:]))
+
+    def asnumpy(self):
+        """
+        Return image data as a (<stack>, angles, radians)
+        read-only array view.
+
+        :return: read-only ndarray view
+        """
+
+        view = self._data.view()
+        view.flags.writeable = False
+        return view
+
+    def copy(self):
+        return self.__class__(self._data.copy())
+
+    def __str__(self):
+        return f"Sinogram(n_images = {self.n}, n_angles = {self.n_points}, n_radial_points = {self.n_radial_points})"
+
+    def backproject(self, angles):
+        """
+        Back Projection Method for a single stack of lines.
+
+        :param angles: np.ndarray
+            1D array of angles in radians. Each entry in the array corresponds to a different number of angles which are used to reconstruct the image.
+        :return: Aspire Image
+            An Image object containing the original stack size with a newly reconstructed numpy array of the images. Expected return shape should be (n_images, n_angles, n_radial_points)
```

Comment:
I think you have the return shape wrong here. Should it be `(..., n_radial_points, n_radial_points)`?

Reply:
Looks good. Thanks.